src/main/modules/pcq/index.ts

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import crypto from 'crypto';
+
+import { AxiosError } from 'axios';
+import config from 'config';
+import { Application, Response } from 'express';
+import { v4 as uuid } from 'uuid';
+import { LoggerInstance } from 'winston';
+
+import { AppRequest } from '../../app/controller/AppRequest';
+
+import { PCQController, PcqController } from './controller';
+import { PcqParameters } from './definitions';
+import { PCQRoute, PcqRoute } from './routes';
+import { PCQService, PcqService } from './service';
+
+export class PcqProvider {
+  private isEnabled = false;
+  private logger: LoggerInstance | Record<string, never> = {};
+
+  private algorithm = 'aes-256-gcm';
+  private bufferSize = 16;
+  private iv = Buffer.alloc(this.bufferSize, 0); // Initialization vector.
+  private keyLen = 32;
+  private route: PcqRoute;
+  service: PcqService;
+  controller: PcqController;
+
+  constructor() {
+    this.service = PCQService;
+    this.controller = PCQController;
+    this.route = PCQRoute;
+  }
+
+  private savePcqId(req: AppRequest, pcqId: string): Promise<void> {
+    return new Promise(resolve => {
+      (async () => {
+        await this.resetPcqId(req);
+        if (req.session.applicationSettings) {
+          req.session.applicationSettings.pcqId = pcqId;
+        } else {
+          req.session.applicationSettings = {
+            pcqId,
+          };
+        }
+        return req.session.save(resolve);
+      })();
+    });
+  }
+
+  getPcqId(req: AppRequest): string | null {
+    return req.session?.applicationSettings?.pcqId;
+  }
+
+  private resetPcqId(req: AppRequest): Promise<void> {
+    return new Promise(resolve => {
+      if (req.session?.applicationSettings?.pcqId) {
+        req.session.applicationSettings.pcqId = null;
+        return req.session.save(resolve);
+      } else {
+        resolve();
+      }
+    });
+  }
+
+  async enable(app: Application): Promise<void> {
+    this.isEnabled = await this.isComponentEnabled();
+    if (this.isEnabled) {
+      this.route.enable(app);
+    }
+  }
+
+  async isComponentEnabled(): Promise<boolean> {
+    const pcqEnabled = config.get('services.equalityAndDiversity.pcqEnabled');
+    const isEnabled = pcqEnabled ? pcqEnabled === 'true' : false;
+    return new Promise(resolve => {
+      resolve(isEnabled);
+    });
+  }
+
+  async init(appRequest: AppRequest): Promise<void> {
+    this.logger = appRequest.locals.logger;
```

Comment:
why is init required for PCQ? I don't think its required at all.

Reply:
Just to initialise the logger 
I am calling this method  as part of pcq module so should not be an issue

Reply:
You could initialise logger in some other singleton method. Try if you can initialise it elsewhere?